src/app/Shared/Redux/Filters/TopologyFilterSlice.tsx

```diff
@@ -0,0 +1,269 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { EnvironmentNode, NodeType, TargetNode } from '@app/Topology/typings';
+import { createAction, createReducer } from '@reduxjs/toolkit';
+import { ReducerWithInitialState } from '@reduxjs/toolkit/dist/createReducer';
+import { getPersistedState } from '../utils';
+
+const _version = '1';
+
+export enum TopologyFilterAction {
+  CATEGORY_TYPE_UPDATE = 'topology-category-type/update',
+  CATEGORY_UPDATE = 'topology-category/update',
+  FILTER_ADD = 'topology-filter/add',
+  FILTER_DELETE = 'topology-filter/delete', // Delete a filter in a category
+  FILTER_DELETE_ALL = 'topology-filter/delete-all', // Delete all filters in all categories
+  CATEGORY_FILTERS_DELETE = 'topology-filter/delete-category', // Delete all filters of the same category
+}
+
+export const enumValues = new Set(Object.values(TopologyFilterAction));
+
+export const topologyUpdateCategoryTypeIntent = createAction(
+  TopologyFilterAction.CATEGORY_TYPE_UPDATE,
+  (isGroup: boolean) => ({
+    payload: {
+      isGroup,
+    },
+  })
+);
+
+export const topologyUpdateCategoryIntent = createAction(
+  TopologyFilterAction.CATEGORY_UPDATE,
+  (isGroup: boolean, category: string) => ({
+    payload: {
+      isGroup,
+      category,
+    },
+  })
+);
+
+export const topologyAddFilterIntent = createAction(
+  TopologyFilterAction.FILTER_ADD,
+  (isGroup: boolean, nodeType: NodeType, category: string, value: string) => ({
+    payload: {
+      isGroup,
+      nodeType,
+      category,
+      value,
+    },
+  })
+);
+
+export const topologyDeleteFilterIntent = createAction(
+  TopologyFilterAction.FILTER_DELETE,
+  (isGroup: boolean, nodeType: NodeType, category: string, value: string) => ({
+    payload: {
+      isGroup,
+      nodeType,
+      category,
+      value,
+    },
+  })
+);
+
+export const topologyDeleteAllFiltersIntent = createAction(TopologyFilterAction.FILTER_DELETE_ALL, () => ({
+  payload: {},
+}));
+
+export const topologyDeleteCategoryFiltersIntent = createAction(
+  TopologyFilterAction.CATEGORY_FILTERS_DELETE,
+  (isGroup: boolean, nodeType: NodeType, category: string) => ({
+    payload: {
+      isGroup,
+      nodeType,
+      category,
+    },
+  })
+);
+
+export interface TopologyFilters {
+  isGroup: boolean;
+  groupFilters: {
+    category: string;
+    filters: {
+      [nodeType: string]: {
+        Name: string[];
+        Label: string[];
+      };
+    };
+  };
+  targetFilters: {
+    category: string;
+    filters: {
+      [nodeType: string]: {
+        Alias: string[];
+        ConnectionUrl: string[];
+        s;
```

Comment:
What is this `s` for?

Reply:
Nope thats the typo. I should not be there. Must be during typing XD